From 04f232ce58422f60b1f1e14c3aedf1e021914cb5 Mon Sep 17 00:00:00 2001
From: Sergio Martins <sergio.martins@kdab.com>
Date: Tue, 10 Jul 2018 12:05:26 +0100
Subject: [PATCH 16/32] Fix Application flags, they can't be bigger than 31

Since they are bitshifted.
Fixed tooltips getting a very big height and possibly other random
bugs since attribute testing was broken.

Change-Id: Ia06c5b583787b69fd44ad73f3e655e73ee7c0197
Reviewed-on: https://codereview.kdab.com/59641
Reviewed-by: Nathan Collins <nathan.collins@kdab.com>
Reviewed-by: Michael Hughes <mhughes@amazon.com>
Reviewed-on: https://codereview.kdab.com/69430
Reviewed-by: Chris Galvan <chgalvan@amazon.com>

# Conflicts:
#	src/corelib/global/qnamespace.h
---
 src/corelib/global/qnamespace.h | 60 ++++++++++-----------------------
 1 file changed, 18 insertions(+), 42 deletions(-)

diff --git a/src/corelib/global/qnamespace.h b/src/corelib/global/qnamespace.h
index e39007f861..6dafa0148e 100644
--- a/src/corelib/global/qnamespace.h
+++ b/src/corelib/global/qnamespace.h
@@ -1,6 +1,6 @@
 /****************************************************************************
 **
-** Copyright (C) 2020 The Qt Company Ltd.
+** Copyright (C) 2016 The Qt Company Ltd.
 ** Contact: https://www.qt.io/licensing/
 **
 ** This file is part of the QtCore module of the Qt Toolkit.
@@ -50,10 +50,10 @@ QT_BEGIN_NAMESPACE
 
 #if !defined(Q_QDOC) && !defined(Q_MOC_RUN)
 struct QMetaObject;
-const QMetaObject *qt_getQtMetaObject() noexcept; // defined in qobject.h (which can't be included here)
+const QMetaObject *qt_getQtMetaObject() Q_DECL_NOEXCEPT; // defined in qobject.h (which can't be included here)
 #define QT_Q_ENUM(ENUM) \
-    inline const QMetaObject *qt_getEnumMetaObject(ENUM) noexcept { return qt_getQtMetaObject(); } \
-    inline Q_DECL_CONSTEXPR const char *qt_getEnumName(ENUM) noexcept { return #ENUM; }
+    inline const QMetaObject *qt_getEnumMetaObject(ENUM) Q_DECL_NOEXCEPT { return qt_getQtMetaObject(); } \
+    inline Q_DECL_CONSTEXPR const char *qt_getEnumName(ENUM) Q_DECL_NOEXCEPT { return #ENUM; }
 #define QT_Q_FLAG(ENUM) QT_Q_ENUM(ENUM)
 #else
 #define QT_Q_ENUM Q_ENUM
@@ -1227,8 +1227,7 @@ public:
     enum TextFormat {
         PlainText,
         RichText,
-        AutoText,
-        MarkdownText
+        AutoText
     };
 
     enum AspectRatioMode {
@@ -1275,16 +1274,14 @@ public:
     enum DateFormat {
         TextDate,      // default Qt
         ISODate,       // ISO 8601
-#if QT_DEPRECATED_SINCE(5, 15)
-        SystemLocaleDate Q_DECL_ENUMERATOR_DEPRECATED_X("Use QLocale"),
-        LocalDate Q_DECL_ENUMERATOR_DEPRECATED_X("Use QLocale") = 2, // i.e. SystemLocaleDate
-        LocaleDate Q_DECL_ENUMERATOR_DEPRECATED_X("Use QLocale"),
-        SystemLocaleShortDate Q_DECL_ENUMERATOR_DEPRECATED_X("Use QLocale"),
-        SystemLocaleLongDate Q_DECL_ENUMERATOR_DEPRECATED_X("Use QLocale"),
-        DefaultLocaleShortDate Q_DECL_ENUMERATOR_DEPRECATED_X("Use QLocale"),
-        DefaultLocaleLongDate Q_DECL_ENUMERATOR_DEPRECATED_X("Use QLocale"),
-#endif
-        RFC2822Date = 8, // RFC 2822 (+ 850 and 1036 during parsing)
+        SystemLocaleDate, // deprecated
+        LocalDate = SystemLocaleDate, // deprecated
+        LocaleDate,     // deprecated
+        SystemLocaleShortDate,
+        SystemLocaleLongDate,
+        DefaultLocaleShortDate,
+        DefaultLocaleLongDate,
+        RFC2822Date,        // RFC 2822 (+ 850 and 1036 during parsing)
         ISODateWithMs
     };
 
@@ -1411,9 +1408,7 @@ public:
     enum InputMethodQuery {
         ImEnabled = 0x1,
         ImCursorRectangle = 0x2,
-#if QT_DEPRECATED_SINCE(5, 14)
-        ImMicroFocus Q_DECL_ENUMERATOR_DEPRECATED = 0x2,
-#endif
+        ImMicroFocus = 0x2, // deprecated
         ImFont = 0x4,
         ImCursorPosition = 0x8,
         ImSurroundingText = 0x10,
@@ -1540,12 +1535,10 @@ public:
         // Metadata
         FontRole = 6,
         TextAlignmentRole = 7,
+        BackgroundColorRole = 8,
         BackgroundRole = 8,
+        TextColorRole = 9,
         ForegroundRole = 9,
-#if QT_DEPRECATED_SINCE(5, 13) // ### Qt 6: remove me
-        BackgroundColorRole Q_DECL_ENUMERATOR_DEPRECATED = BackgroundRole,
-        TextColorRole Q_DECL_ENUMERATOR_DEPRECATED = ForegroundRole,
-#endif
         CheckStateRole = 10,
         // Accessibility
         AccessibleTextRole = 11,
@@ -1586,12 +1579,9 @@ public:
         MatchContains = 1,
         MatchStartsWith = 2,
         MatchEndsWith = 3,
-#if QT_DEPRECATED_SINCE(5, 15)
-        MatchRegExp Q_DECL_ENUMERATOR_DEPRECATED_X("MatchRegExp is deprecated. Use MatchRegularExpression instead") = 4,
-#endif
+        MatchRegExp = 4,
         MatchWildcard = 5,
         MatchFixedString = 8,
-        MatchRegularExpression = 9,
         MatchCaseSensitive = 16,
         MatchWrap = 32,
         MatchRecursive = 64
@@ -1651,7 +1641,7 @@ public:
         TitleBarArea    // For move
     };
 
-#if defined(Q_COMPILER_CONSTEXPR)
+#if defined(Q_COMPILER_CLASS_ENUM) && defined(Q_COMPILER_CONSTEXPR)
     enum class Initialization {
         Uninitialized
     };
@@ -1768,18 +1758,6 @@ public:
         ChecksumItuV41
     };
 
-    enum class HighDpiScaleFactorRoundingPolicy {
-        Unset,
-        Round,
-        Ceil,
-        Floor,
-        RoundPreferFloor,
-        PassThrough
-    };
-
-    // QTBUG-48701
-    enum ReturnByValueConstant { ReturnByValue }; // ### Qt 7: Remove me
-
 #ifndef Q_QDOC
     // NOTE: Generally, do not add QT_Q_ENUM if a corresponding Q_Q_FLAG exists.
     QT_Q_ENUM(ScrollBarPolicy)
@@ -1806,7 +1784,6 @@ public:
     QT_Q_FLAG(Alignment)
     QT_Q_ENUM(TextFlag)
     QT_Q_FLAG(Orientations)
-    QT_Q_FLAG(SplitBehavior)
     QT_Q_FLAG(DropActions)
     QT_Q_FLAG(Edges)
     QT_Q_FLAG(DockWidgetAreas)
@@ -1866,7 +1843,6 @@ public:
     QT_Q_ENUM(MouseEventSource)
     QT_Q_FLAG(MouseEventFlag)
     QT_Q_ENUM(ChecksumType)
-    QT_Q_ENUM(HighDpiScaleFactorRoundingPolicy)
     QT_Q_ENUM(TabFocusBehavior)
 #endif // Q_DOC
 
-- 
2.28.0.windows.1

